From 486a855894935b0c88b88f04174fb4d4b3edf6b3 Mon Sep 17 00:00:00 2001
From: Tobias Heider <tobias.heider@canonical.com>
Date: Tue, 10 Dec 2024 13:04:28 +0100
Subject: [PATCH 39/45] UBUNTU: SAUCE: HACK: Disable speakers by default

Audio support on X Elite is currently experimental and comes
with a risk of permanently damaging hardware. Add an explicit kernel
command line flag to make sure it doesn't get enabled by accident.

Signed-off-by: Tobias Heider <tobias.heider@canonical.com>
---
 sound/soc/qcom/x1e80100.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sound/soc/qcom/x1e80100.c b/sound/soc/qcom/x1e80100.c
index 8eb57fc12f0d..e6f189c8f886 100644
--- a/sound/soc/qcom/x1e80100.c
+++ b/sound/soc/qcom/x1e80100.c
@@ -15,6 +15,9 @@
 #include "qdsp6/q6dsp-common.h"
 #include "sdw.h"
 
+bool i_accept_the_danger = 0;
+module_param(i_accept_the_danger, bool, 0644);
+
 struct x1e80100_snd_data {
 	bool stream_prepared[AFE_PORT_MAX];
 	struct snd_soc_card *card;
@@ -193,6 +196,9 @@ static int x1e80100_platform_probe(struct platform_device *pdev)
 	struct device *dev = &pdev->dev;
 	int ret;
 
+	if (!i_accept_the_danger)
+		return -EINVAL;
+
 	card = devm_kzalloc(dev, sizeof(*card), GFP_KERNEL);
 	if (!card)
 		return -ENOMEM;
-- 
2.51.0

